internal/ethereum/blocklistener.go

```diff
@@ -185,6 +194,8 @@ func (bl *blockListener) listenLoop() {
 				log.L(bl.ctx).Errorf("Failed to establish new block filter: %s", err.Message)
 				failCount++
 				continue
+			} else {
+				close(bl.blockFilterEstablished)
```

Comment:
need a check here to make sure closing the channel is only attempted once.

Reply:
done now